provider/pact-jvm-provider-junit/src/test/groovy/au/com/dius/pact/provider/junit/descriptions/DescriptionGeneratorTest.groovy

```diff
@@ -0,0 +1,62 @@
+package au.com.dius.pact.provider.junit.descriptions
+
+import au.com.dius.pact.core.model.RequestResponseInteraction
+import au.com.dius.pact.core.model.RequestResponsePact
+import au.com.dius.pact.core.model.BrokerUrlSource
+import au.com.dius.pact.core.model.DirectorySource
+import au.com.dius.pact.core.model.Request
+import au.com.dius.pact.core.model.Response
+import au.com.dius.pact.core.model.Provider
+import au.com.dius.pact.core.model.Consumer
+import au.com.dius.pact.core.model.ProviderState
+import au.com.dius.pact.provider.junit.target.HttpTarget
+import au.com.dius.pact.provider.junit.target.Target
+import au.com.dius.pact.provider.junit.target.TestTarget
+import org.junit.runners.model.TestClass
+import spock.lang.Specification
+
+class DescriptionGeneratorTest extends Specification {
+
+  @SuppressWarnings('PublicInstanceField')
+  class DescriptionGeneratorTestClass {
+    @TestTarget
+    public final Target target = new HttpTarget(8332)
+  }
+
+  private clazz
+  private interaction
+  private pact
+
+  def setup() {
+    clazz = new TestClass(DescriptionGeneratorTestClass)
+  }
+
+  def 'when BrokerUrlSource tests description includes tag if present'() {
+    def interaction = new RequestResponseInteraction('Interaction 1',
+            [ new ProviderState('Test State') ], new Request(), new Response())
+    def pact = new RequestResponsePact(new Provider(), new Consumer(), [ interaction ])
+
+    expect:
+    def pactSource =  new BrokerUrlSource('url', 'url', [:], [:], tag)
+    def generator = new DescriptionGenerator(clazz, pact, pactSource)
+    description == generator.generate(interaction).methodName
+
+    where:
+    tag      | description
+    'master' | '[tag:master] consumer - Upon Interaction 1'
```

Comment:
Is 'consumer' the name of the consumer? To me, it would make more sense if the `[tag:master]` came after the consumer name - but that's just a very small preference.

Reply:
I personally agree, appreciate the suggestion. Will apply change